File: TestSessionsClient.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Net.Http.Json;
using System.Text.Json;
using System.Threading.Tasks;

public class TestSessionsClient
{
    private readonly HttpClient _httpClient;
    private readonly string _baseUrl;
    private readonly string _apiKey;

    public TestSessionsClient(string baseUrl, string apiKey)
    {
        _baseUrl = baseUrl.TrimEnd('/');
        _apiKey = apiKey;
        _httpClient = new HttpClient();
    }

    private void AddApiKeyHeader(HttpRequestMessage request)
    {
        request.Headers.Remove("x-api-key");
        request.Headers.Add("x-api-key", _apiKey);
    }

    public async Task<List<DeviceSession>> GetAllSessionsAsync()
    {
        var request = new HttpRequestMessage(HttpMethod.Get, $"{_baseUrl}/test-sessions");
        AddApiKeyHeader(request);
        var response = await _httpClient.SendAsync(request);
        response.EnsureSuccessStatusCode();
        var sessions = await response.Content.ReadFromJsonAsync<List<DeviceSession>>();
        return sessions ?? new List<DeviceSession>();
    }

    public async Task<DeviceSession?> PostSessionTreeAsync(string deviceKey, object tree)
    {
        var request = new HttpRequestMessage(HttpMethod.Post, $"{_baseUrl}/test-sessions/{deviceKey}")
        {
            Content = JsonContent.Create(tree)
        };
        AddApiKeyHeader(request);
        var response = await _httpClient.SendAsync(request);
        response.EnsureSuccessStatusCode();
        var session = await response.Content.ReadFromJsonAsync<DeviceSession>();
        return session;
    }

    public async Task<JsonElement?> GetSessionTreeAsync(string deviceKey)
    {
        var request = new HttpRequestMessage(HttpMethod.Get, $"{_baseUrl}/test-sessions/{deviceKey}");
        AddApiKeyHeader(request);
        var response = await _httpClient.SendAsync(request);
        if (response.StatusCode == System.Net.HttpStatusCode.NotFound)
            return null;
        response.EnsureSuccessStatusCode();
        var tree = await response.Content.ReadFromJsonAsync<JsonElement>();
        return tree;
    }
}

public class DeviceSession
{
    public string deviceKey { get; set; } = string.Empty;
    public JsonElement tree { get; set; }
} 

File: testSessionsClient.ts
export interface DeviceSession {
  deviceKey: string;
  tree: any;
}

export class TestSessionsClient {
  private baseUrl: string;
  private apiKey: string;

  constructor(baseUrl: string, apiKey: string) {
    this.baseUrl = baseUrl.replace(/\/$/, '');
    this.apiKey = apiKey;
  }

  private getHeaders(): HeadersInit {
    return {
      'Content-Type': 'application/json',
      'x-api-key': this.apiKey,
    };
  }

  async getAllSessions(): Promise<DeviceSession[]> {
    const res = await fetch(`${this.baseUrl}/test-sessions`, {
      headers: this.getHeaders(),
    });
    if (!res.ok) throw new Error('Failed to fetch sessions');
    return res.json();
  }

  async postSessionTree(deviceKey: string, tree: any): Promise<DeviceSession> {
    const res = await fetch(`${this.baseUrl}/test-sessions/${deviceKey}`, {
      method: 'POST',
      headers: this.getHeaders(),
      body: JSON.stringify(tree),
    });
    if (!res.ok) throw new Error('Failed to post session tree');
    return res.json();
  }

  async getSessionTree(deviceKey: string): Promise<any | null> {
    const res = await fetch(`${this.baseUrl}/test-sessions/${deviceKey}`, {
      headers: this.getHeaders(),
    });
    if (res.status === 404) return null;
    if (!res.ok) throw new Error('Failed to fetch session tree');
    return res.json();
  }
} 